test/integration/helpers.bash

```diff
@@ -88,24 +88,7 @@ function swarm_join() {
 		SWARM_JOIN_PID[$i]=$!
 		((++i))
 	done
-	wait_until_swarm_joined $i
-}
-
-# Wait until a swarm instance joins the cluster.
-# Parameter $1 is number of nodes to check.
-function wait_until_swarm_joined {
-	local attempts=0
-	local max_attempts=10
-
-	until [ $attempts -ge $max_attempts ]; do
-		run docker -H $SWARM_HOST info
-		if [[ "${lines[3]}" == *"Nodes: $1"* ]]; then
-			break
-		fi 
-		echo "Checking if joined successfully for the $((++attempts)) time" >&2
-		sleep 1
-	done
-	[[ $attempts -lt $max_attempts ]]
+	retry 30 1 [ `docker_swarm info | grep "Nodes: $i"` -eq 0 ]
```

Comment:
it's testing the exit code of the expression in backticks.
`if [[ `echo 2 | grep -q 2` -eq 0 ]]; then echo yes; else echo no; fi` shows yes and 
`if [[ `echo 2 | grep -q 2` -eq 1 ]]; then echo yes; else echo no; fi` shows no for me.

Maybe I should add `-q` to it.